### DISCUSSION TITLE
how does zsh-autocomplete limit zsh completion initial listing ?

### DISCUSSION THREAD

**Discussion by lafleurdeboum** [marlonrichert/zsh-autocomplete#560] [Q&A]
In [this SO question](https://stackoverflow.com/questions/75762775/how-do-i-limit-zsh-completion-initial-listings-length), I'm looking for ways to force zsh's initial listing on ambiguous completion. Just like if I replaced the first tab completion's pager by a call to head.

I tried zsh-autocomplete and it does that very nicely ! It's just that I'm not interested in a complete-as-you-type system, I'd rather update [my own zshrc](https://git.presentmatter.one/lafleur/dotfiles/src/branch/master/src/.zshrc) to implement that feature.

So the question is - how does zsh-autocomplete manage to do that ? @marlonrichert you mentioned a subprocess that trims the completions to fit onto the screen. If you (or anyone else) would care to give some details on that process, I'd be very grateful.

What I can't figure at the moment is how that subprocess triggers a completion and reads the results. I tried reading through the code, but apparently was quickly mistaken.

**Comment by marlonrichert** [comment]
It happens in the function [`.autocomplete.async.compadd`](https://github.com/marlonrichert/zsh-autocomplete/blob/86ffb11c7186664a71fd36742f3148628c4b85cb/scripts/.autocomplete.async#L494) and it's actually quite complex. Let me know if you have any about any lines there in particular. I can add more comments to the code. 🙂
